ofrak_core/ofrak/core/entropy/entropy.c

```diff
@@ -1,17 +1,28 @@
 #include <stddef.h>   // size_t, NULL
 #include <inttypes.h> // uint8_t, uint32_t
 #include <math.h>     // floor, log2
+#include <Python.h>
```

Comment:
Good point about the dependency, I'll make sure to add it to the Dockerstub.

I don't love that you can make the C extension optional, but if it fails it is completely silent. I would hope for at least a small warning. Also note that I had to update setuptools because the version I had was hard failing even with the `optional` flag set True (this was when I was intentionally building a bad version of the extension to test the fallback code). This isn't ideal, but not a huge deal, just thought it worth noting.

I like the testing idea, I'll implement that. I was thinking that our tests don't check for the successful C version, so that would cover that.

Reply:
Actually since our base image for the docker file is `python:3.7` which includes the python dev headers, it's not needed in the Dockerstub.

Reply:
Sounds good. I assume there is no way to `pip install` the headers, or to indicate their necessity for a `pip install` user.